src/Microsoft.Azure.WebJobs.Host/Timers/IWebJobsExceptionHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host.Timers
+{
+    /// <summary>
+    /// Represents an unhandled exception handler.
+    /// </summary>
+    public interface IWebJobsExceptionHandler
+    {
+        /// <summary>
+        /// Called during host creation. Allows a reference to the host.
+        /// </summary>
+        /// <param name="host">The JobHost.</param>
+        void Initialize(JobHost host);
```

Comment:
Fabio also suggested a factory for scenarios like this. I'm going to leave it as-is for the moment and revisit for improvements later.

---
In reply to: [77075564](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r77075564) [](ancestors = 77075564)